### PR TITLE
zh-translation: docs/concepts/services-networking/service-topology.md

### PR DESCRIPTION
**Pull request by helight**
zh-translation: docs/concepts/services-networking/service-topology.md
ref: https://github.com/k8smeetup/website-tasks/issues/3663

### CODE REVIEW

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+may designate groups of Nodes that are "closer" and "farther" from one another,
+using whatever metric makes sense for that operator's requirements. For many
+operators in public clouds, for example, there is a preference to keep service
+traffic within the same zone, because interzonal traffic has a cost associated
+with it, while intrazonal traffic does not. Other common needs include being able
+to route traffic to a local Pod managed by a DaemonSet, or keeping traffic to
+Nodes connected to the same top-of-rack switch for the lowest latency.
+-->
+
+## 介绍 {#introduction}
```
[the remark this answers is not in the document]
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: got that

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
+---
+reviewers:
+- johnbelamaric
+- imroc
+title: Service 拓扑
+feature:
+  title: Service 拓扑
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service-topology.md]: Service Topology 是一个专有名词，为什么只翻译其中一个单词？

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+operators in public clouds, for example, there is a preference to keep service
+traffic within the same zone, because interzonal traffic has a cost associated
+with it, while intrazonal traffic does not. Other common needs include being able
+to route traffic to a local Pod managed by a DaemonSet, or keeping traffic to
+Nodes connected to the same top-of-rack switch for the lowest latency.
+-->
+
+## 介绍 {#introduction}
+
+默认情况下，发往 `ClusterIP` 或者 `NodePort` 服务的流量可能会被路由到任意一个服务后端的地址上。从 Kubernetes 1.7 开始，可以将“外部”流量路由到节点上运行的 pod 上，但不支持 `ClusterIP` 服务，并且更复杂的拓扑 &mdash; 比如分区路由 &mdash; 也还不支持。通过允许 `Service` 创建器根据 `Node` 标签为源 `Node` 和目的 `Node` 定义流量路由策略，`Service` 拓扑特性实现了服务流量的路由。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service-topology.md]:
```suggestion
默认情况下，发往 `ClusterIP` 或者 `NodePort` 服务的流量可能会被路由到任意一个服务后端的地址上。从 Kubernetes 1.7 开始，可以将“外部”流量路由到节点上运行的 pod 上，但不支持 `ClusterIP` 服务。更复杂的拓扑 &mdash; 比如分区路由 &mdash; 也还不支持。通过允许 `Service` 创建者根据源 `Node` 和目的 `Node` 的标签来定义流量路由策略，`Service` 拓扑特性解决了上述问题。
```

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+通过对源 `Node` 和目的 `Node` 标签的匹配，运营者可以使用任何符合运营者要求的度量来指定彼此“更近”和“更远”的节点组。例如，对于在公有云上的运营者来说，更偏向于把流量控制在同一区域内，应为区域间的流量是有费用成本的，而区域内的流量没有。其它常用需要包括把流量路由到由 `DaemonSet` 管理的本地 Pod 上，或者把保持流量在链接同一机架交换机的 `Node` 上，以获得低延时。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service-topology.md]:
```suggestion
通过对源 `Node` 和目的 `Node` 标签的匹配，运营者可以使用任何符合运营者要求的度量值来设定彼此“较近”和“较远”的节点组。例如，对于在公有云上的运营者来说，更偏向于把流量控制在同一区域内，因为区域间的流量是有费用成本的，而区域内的流量没有。其它常见的需求还包括把流量路由到由 `DaemonSet` 管理的本地 Pod 上，或者把保持流量在连接同一机架交换机的 `Node` 上，以获得低延时。
```

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+routing:
+
+   * Kubernetes 1.17 or later
+   * Kube-proxy running in iptables mode or IPVS mode
+   * Enable [Endpoint Slices](/zh/docs/concepts/services-networking/endpoint-slices/)
+-->
+
+## 前提条件 {#prerequisites}
+
+为了启用拓扑感知服务路由功能，必须要有以下一些前提条件：
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service-topology.md]:
```suggestion
为了启用拓扑感知服务路由功能，必须要满足以下一些前提条件：
```

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+   * Kubernetes 的版本要 >= 1.17
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service-topology.md]:
```suggestion
   * Kubernetes 的版本不低于 1.17
```

File: content/zh/docs/concepts/services-networking/service-topology.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+  none are available within this zone:
+  `["topology.kubernetes.io/zone", "*"]`.
+-->
+
+如果 `topologyKeys` 没有指定或者为空，就没有启用这个拓扑功能。
+
+一个集群中，其 `Node` 的标签被打为其主机名，区域名和地区名。那么就可以设置 `Service` 的 `topologyKeys` 的值，像下面的做法一样定向流量了。
+
+* 只定向到同一个 `Node` 上的端点，`Node` 上没有端点存在时就失败：配置 `["kubernetes.io/hostname"]`。
+* 偏向定向到同一个 `Node`  上的端点，回退同一区域的端点上，然后是同一地区，其它情况下就失败：配置 `["kubernetes.io/hostname",  "topology.kubernetes.io/zone", "topology.kubernetes.io/region"]`。这或许很有用，例如，数据局部很重要的情况下。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service-topology.md]:
```suggestion
* 偏向定向到同一个 `Node`  上的端点，回退同一区域的端点上，然后是同一地区，其它情况下就失败：配置 `["kubernetes.io/hostname",  "topology.kubernetes.io/zone", "topology.kubernetes.io/region"]`。这或许很有用，例如，数据局部性很重要的情况下。
```

File: content/zh/docs/concepts/services-networking/service-topology.md (hunk 2 above)
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: 这个你建议是不翻译吗？也是纠结了半天，最终只翻译了一半。

File: content/zh/docs/concepts/services-networking/service-topology.md (hunk 3 above)
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: accept

File: content/zh/docs/concepts/services-networking/service-topology.md (hunk 4 above)
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: accept

File: content/zh/docs/concepts/services-networking/service-topology.md (hunk 5 above)
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: accept

File: content/zh/docs/concepts/services-networking/service-topology.md (hunk 6 above)
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: accept

File: content/zh/docs/concepts/services-networking/service-topology.md (hunk 7 above)
Comment (helight) [reply content/zh/docs/concepts/services-networking/service-topology.md]: accept，多谢多谢

**Comment by rudy2steiner** [comment]
I signed it

**Pull request merged**